delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/multifetch/MultiFetchSuite.scala

```diff
@@ -0,0 +1,86 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.MultiFetchResponse.Result.{AuthorizationFailed, NotFound, Success}
+import ch.epfl.bluebrain.nexus.delta.sdk.multifetch.model.{MultiFetchRequest, MultiFetchResponse}
+import ch.epfl.bluebrain.nexus.delta.sdk.permissions.Permissions
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.Fixtures
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ResourceRef.Latest
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{Identity, Label, ProjectRef}
+import ch.epfl.bluebrain.nexus.testkit.TestHelpers
+import ch.epfl.bluebrain.nexus.testkit.bio.BioSuite
+import monix.bio.UIO
+
+class MultiFetchSuite extends BioSuite with TestHelpers with Fixtures {
+
+  implicit private val subject: Subject = Identity.User("user", Label.unsafe("realm"))
+  implicit private val caller: Caller   = Caller.unsafe(subject)
+
+  private val project1 = ProjectRef.unsafe("org", "proj1")
+  private val project2 = ProjectRef.unsafe("org", "proj2")
+
+  private val permissions = Set(Permissions.resources.read)
+  private val aclCheck    = AclSimpleCheck((subject, project1, permissions)).runSyncUnsafe()
+
+  private val successId      = nxv + "success"
+  private val successContent =
+    ResourceGen.jsonLdContent(successId, project1, jsonContentOf("resources/resource.json", "id" -> successId))
+  private val notFoundId     = nxv + "not-found"
+  private val unauthorizedId = nxv + "unauthorized"
+
+  private def fetchResource =
+    (input: MultiFetchRequest.Input) => {
+      input match {
+        case MultiFetchRequest.Input(Latest(`successId`), `project1`) =>
+          UIO.some(successContent)
+        case _                                                        => UIO.none
+      }
+    }
+
+  private val multiFetch = MultiFetch(
+    aclCheck,
+    fetchResource
+  )
+
+  private val request = MultiFetchRequest(
+    ResourceRepresentation.NTriples,
+    Input(Latest(successId), project1),
+    Input(Latest(notFoundId), project1),
+    Input(Latest(unauthorizedId), project2)
+  )
+
+  test("Return the response matching the user acls") {
+
+    val expected = MultiFetchResponse(
+      ResourceRepresentation.NTriples,
+      NonEmptyList.of(
+        Success(Latest(successId), project1, successContent),
+        NotFound(Latest(notFoundId), project1),
+        AuthorizationFailed(Latest(unauthorizedId), project2)
+      )
+    )
+
+    multiFetch(request).assert(expected)
+  }
+
+  test("Return only unauthorized for a user with no access") {
+    val expected = MultiFetchResponse(
+      ResourceRepresentation.NTriples,
+      NonEmptyList.of(
+        AuthorizationFailed(Latest(successId), project1),
+        AuthorizationFailed(Latest(notFoundId), project1),
+        AuthorizationFailed(Latest(unauthorizedId), project2)
+      )
+    )
+
+    multiFetch(request)(Caller.Anonymous).assert(expected)
+  }
```

Comment:
Consider having more, less broad tests. For example, returning not found is probably one test. There are more tests but they are more straightforward and point to the defect more directly

Reply:
Multi-fetch is supposed to return the resources in the order they are requested whether it is a success or not.
That is why I took this approach to validate also this part.

Reply:
Yeah I'm just saying that these tests are basically testing multiple things

Another way of writing these tests would be to have:

- A resource is found
- A resource is missing
- A resource has no authorization
- Resources are in order
- Resources are returned as type X

Reply:
I would agree if the complexity was higher but here it is simple